torch/utils/benchmark/utils/cpp_jit.py

```diff
@@ -0,0 +1,44 @@
+"""JIT C++ strings into executables."""
+import os
+import threading
+from typing import List, Optional
+
+import torch
+from torch.utils.benchmark.utils._stubs import CallgrindModuleType
+from torch.utils import cpp_extension
+
+
+LOCK = threading.Lock()
+SOURCE_ROOT = os.path.split(os.path.abspath(__file__))[0]
+
+if hasattr(torch.__config__, "_cxx_flags"):
+    CXX_FLAGS = torch.__config__._cxx_flags().strip().split()
+    if "-g" not in CXX_FLAGS:
+        CXX_FLAGS.append("-g")
+else:
+    # FIXME: Remove when back testing is no longer required.
+    CXX_FLAGS = ["-O2", "-fPIC", "-g"]
+
+EXTRA_INCLUDE_PATHS: List[str] = [os.path.join(SOURCE_ROOT, "valgrind_wrapper")]
+CONDA_PREFIX = os.getenv("CONDA_PREFIX")
+if CONDA_PREFIX is not None:
+    # load will automatically search /usr/include, but not conda include.
+    EXTRA_INCLUDE_PATHS.append(os.path.join(CONDA_PREFIX, "include"))
+
+
```

Comment:
It would probably be nice to have a Note explaining at a high level what's going on here

Reply:
I understand that this is "temporary" stuff but because it interacts nontrivially with some code that isn't in PyTorch itself, it will be harder for other people to figure out how this relates to the bigger picture. A note here will help a lot.

Reply:
No, it's a valid point. Check out `BACK_TESTING_NOTE:` and let me know if it seems reasonable.